### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by PEZO19** [Skipperlla/rn-swiper-list#21]
I have just updated to 2.0.2 as I have realized that there is a memory leak in the previous version ("rn-tinder-card": "1.6.0",), even if manually calling `gc`.

However, it seems to me 2.0.2 also does have this problem. When looking at the demo mp4 in the README: it starts from 209MB and after 30 sec it has 214MB. The problematic thing is that using `swipeBack` and any swipe again clearly leads to more and more memory too.

@Skipperlla 
Any ideas where to start? I'd like to help with this as I need functionality provided by this lib asap. :)

**Comment by PEZO19** [comment]
After doing some investigation I assume that the problem might be related to `Reanimated`, but it is not clear to me which exact part of it. Maybe `useSharedValue` is the problem, check here: https://github.com/software-mansion/react-native-reanimated/issues/5800#issuecomment-2032631035

**Comment by PEZO19** [comment]
Now, after further investigation I suspect that the main issue with the previous version (`rn-tinder-cards`) might be not `useSharedValue` but `useAnimatedStyle` which leaks in Reanimated 2, but was fixed Reanimated 3. Regarding `rn-swiper-list` and [this comment](https://github.com/Skipperlla/rn-swiper-list/issues/21#issue-2353533059): garbage collector seems to kick in after 20 extra MBs and solves the problem. That did not happen `rn-tinder-cards` and earlier I did not realize that I have to wait that much extra memory consumption with `rn-swiper-list`. Seems fine, sorry for bothering. If there is something related I find, I'll post here.
